spectrochempy/analysis/decomposition/iris.py

```diff
@@ -385,6 +396,9 @@ def __init__(
         warm_start=False,
         **kwargs,
     ):
+        if _osqp_is_not_available():
+            return
```

Comment:
done (I wanted to make osqp optional but changed my mind: a bit complicated for me because used in examples and user guide :) -- forgot to remove this